api/src/main/java/org/apache/iceberg/Metrics.java

```diff
@@ -37,12 +37,14 @@
   private Map<Integer, Long> columnSizes = null;
   private Map<Integer, Long> valueCounts = null;
   private Map<Integer, Long> nullValueCounts = null;
+  private Map<Integer, Long> nanValueCounts = null;
   private Map<Integer, ByteBuffer> lowerBounds = null;
   private Map<Integer, ByteBuffer> upperBounds = null;
 
   public Metrics() {
   }
 
+  // for temporary backward compatibility, will be removed when all writers support nanValueCounts
```

Comment:
If we intend to remove it, then we should add `@Deprecated` and Javadoc `@deprecated will be removed in 0.12.0; use ... instead`.